runtime/Java/src/org/antlr/v4/runtime/misc/FlexibleHashMap.java

```diff
@@ -15,7 +15,9 @@
 
 /** A limited map (many unsupported operations) that lets me use
  *  varying hashCode/equals.
+ * @deprecated in favor of using standard maps with custom keys.
  */
+@Deprecated
 public class FlexibleHashMap<K,V> implements Map<K, V> {
```

Comment:
Since this is public API I didn't want to remove it right away. But it's basically unused inside the project now. Let me know if I should completely remove it already.